log.go

```diff
@@ -159,46 +159,48 @@ type EventLogger interface {
 	SerializeContext(ctx context.Context) ([]byte, error)
 }
 
-// Logger retrieves an event logger by name
-func Logger(system string) EventLogger {
+var _ EventLogger = Logger("test-logger")
 
-	// TODO if we would like to adjust log levels at run-time. Store this event
-	// logger in a map (just like the util.Logger impl)
+// Logger retrieves an event logger by name
+func Logger(system string) *ZapEventLogger {
 	if len(system) == 0 {
 		setuplog := getLogger("setup-logger")
-		setuplog.Warning("Missing name parameter")
+		setuplog.Error("Missing name parameter")
 		system = "undefined"
 	}
 
 	logger := getLogger(system)
 
-	return &eventLogger{system: system, Logger: *logger}
+	return &ZapEventLogger{system: system, SugaredLogger: *logger}
 }
 
-// eventLogger implements the EventLogger and wraps a go-logging Logger
-type eventLogger struct {
-	logging.Logger
+// ZapEventLogger implements the EventLogger and wraps a go-logging Logger
+type ZapEventLogger struct {
+	zap.SugaredLogger
 
 	system string
```

Comment:
nit: This is usually called `category` in other logging systems, e.g. slf4j, logback, etc. `system` makes assumptions about how and where categories will be used.

Reply:
This was already in place there. Do you want me to change it?

Reply:
Only if you feel like it. I've approved anyway!